src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+
+    // to do private resize and getIndex
+    private static final int INITIAL_CAPACITY = 10;
+    private int capacity;
+    private int size;
+    private Object[] elements;
+
+    ArrayList() {
+        elements = new Object[INITIAL_CAPACITY];
+        capacity = INITIAL_CAPACITY;
+        size = 0;
+    }
+
+    private int getIndex(T value) {
+        int index = -1;
+        int counter = 0;
+        for (Object o : elements) {
+            if (Objects.equals(o,value)) {
+                index = counter;
+                return index;
+            }
+            counter++;
+        }
+        return index;
+    }
```

Comment:
this is used only in `remove()` so just move it there
this also can be simplified and `Objects.equals()` is not allowed

Reply:
there was no such requirement in task description to not use Objects.equals

Reply:
isn't it recommended to make methods simpler by breaking them into smaller ones?

Reply:
tried to remove that method and don't see why one test stops passing. it works fine with that method.

Reply:
done